FIX #31 : Renamed identifiers according to named convention of identifiers of C Language.

Akshit6828 opened the pull request:
Renamed identifiers according  : 
* ``` '_GLB_DIRECTOR_CONFIG_H' ``` has been changed to  ```GLB_DIRECTOR_CONFIG_H```

 * ```'__IPT_GLBREDIRECT_MATCH__'``` has been changed to ```IPT_GLBREDIRECT_MATCH```

elfring reviewed (commented):
* Please avoid a typo in the commit subject.
* Would it be helpful to explain the change another bit in the commit description?
* Are there any [further update candidates](https://github.com/github/glb-director/blob/9e631ff5ed075a6498efc730905b4497cd5bbce7/src/glb-hashing/pdnet.h#L15 "Another update candidate") to consider?

elfring commented:
How do you think about to improve the wording also a bit for the pull request subject?

Akshit6828 commented:
Sure @elfring. I'll update the changes and let you know shortly! Thanks for mentioning the changes precisely.

elfring commented:
Wording “… naming convention of the C language standard”?

Akshit6828 commented:
The commit https://github.com/github/glb-director/pull/122/commits/438eedfbeebbcbbf05c0ef9342c95a8a2163aacb propose changes as explained below to uphold the reserve identifier violation of C language standards

#### Change 1 
Changed [pdnet.h][pdnet.h] with the following changes :
* Identifier  ```_PDNET_H_``` is changed to ```PDNET_H```
#### Change 2
Changed [packet_parsing.h][packet_parsing.h] with the following changes :
* Identifier  ```_PACKET_PARSING_H_``` is changed to ```PACKET_PARSING_H```



[pdnet.h]:https://github.com/github/glb-director/blob/9e631ff5ed075a6498efc730905b4497cd5bbce7/src/glb-hashing/pdnet.h
[packet_parsing.h]:https://github.com/github/glb-director/blob/master/src/glb-hashing/packet_parsing.h

Akshit6828 commented:
@elfring please review the commit messages and proposed changes and let me know if any alterations are needed.

elfring commented:
* Would you like to reconsider the number of commits for such an issue?
* How do you think about to [extend the search for affected source code](http://en.cppreference.com/w/cpp/language/identifiers#In_declarations "Description for the selection of identifiers in C++")?

Akshit6828 commented:
Ok, @elfring. I'll note this feedback and propose changes as described by you as early as possible.

elfring commented:
Is your interest growing for any source code search patterns which I mentioned for some software components?

Akshit6828 commented:
@elfring sorry to say, but I have less idea regarding how to extend the search source code as this was my first contribution to any big project company and that's why I am not aware of practices to extend the search for the source code. 

But I am interested in learning new things. I am also reading the resource provided by you. 
Please correct me if i am not getting your directions.

Akshit6828 commented:
Actually @elfring I am a little confused as changes proposed by me in commits changed the identifiers which were reserved words.
As this [resource][resource] says that  

> "All identifiers that begin with an underscore are always reserved for use as identifiers with file scope in both the ordinary and tag namespaces."

I think that in my proposed changes, the identifiers should begin with an underscore, and the rest of the changes are alright.

Am I correct sir?

[resource]: https://wiki.sei.cmu.edu/confluence/display/c/DCL37-C.+Do+not+declare+or+define+a+reserved+identifier#DCL37-C.Donotdeclareordefineareservedidentifier-NoncompliantCodeExample%28IncludeGuard%29

Akshit6828 commented:
> Actually @elfring I am a little confused as changes proposed by me in commits changed the identifiers which were reserved words.
> As this [resource](https://wiki.sei.cmu.edu/confluence/display/c/DCL37-C.+Do+not+declare+or+define+a+reserved+identifier#DCL37-C.Donotdeclareordefineareservedidentifier-NoncompliantCodeExample%28IncludeGuard%29) says that
> 
> > "All identifiers that begin with an underscore are always reserved for use as identifiers with file scope in both the ordinary and tag namespaces."
> 
> I think that in my proposed changes, the identifiers should begin with an underscore, and the rest of the changes are alright.
> 
> Am I correct sir?

Like: 
* ```GLB_DIRECTOR_CONFIG_H``` -->```_GLB_DIRECTOR_CONFIG_H``` 
* ```IPT_GLBREDIRECT_MATCH```  --> ```_IPT_GLBREDIRECT_MATCH```
*  ```PDNET_H``` ---> ```_PDNET_H```
* ```PACKET_PARSING_H``` --> ```_PACKET_PARSING_H```

elfring commented:
Yes, in principle. - [Your search approach seems to be incomplete so far](https://github.com/github/glb-director/blob/9e631ff5ed075a6498efc730905b4497cd5bbce7/src/glb-hashing/glb_gue.h#L33 "Another update candidate").

Akshit6828 commented:
ok your mean, I need to search more files and identifiers for such issues?

elfring commented:
I propose to increase source code analysis accordingly.
Did you notice where I reported similar open issues?

Akshit6828 commented:
I've gone through the issues and I couldn't see similar issues. 
It means for increasing source code analysis accordingly, I need to check where the same issue exists.

Sir, please correct me if I am wrong in understanding you.
#### or
If you don't mind can you point my work precisely that what changes I should do so that I can understand my work better?

elfring commented:
>… and I couldn't see similar issues.

Do you find the following search approaches interesting?
* https://github.com/github/glb-director/search?q=define
* https://github.com/issues?q=is%3Aissue+%22reserved+identifier+violation%22

Akshit6828 commented:
Ok sir I'll check these search approaches and change the source code accordingly within few days.
Thanks for mentioning them.

Akshit6828 commented:
> 🤔 Would you like to check any source code places besides questionable include guards?

Hi @elfring.  Sorry, I didn't get that. Could you elaborate  your question?

elfring commented:
Are you familiar with [involved software development challenges for include guards](https://en.wikipedia.org/wiki/Include_guard#Difficulties "Selection of a coherent naming scheme")?

Akshit6828 commented:
No, I am not familiar with it. But I'll read the resource and let you know.
